host/src/FunctionsNetHost/Grpc/IncomingGrpcMessageHandler.cs

```diff
@@ -62,11 +62,14 @@ private async Task Process(StreamingMessage msg)
                     }
                 case StreamingMessage.ContentOneofCase.FunctionEnvironmentReloadRequest:
 
-                    Configuration.Reload();
+                    var envReloadRequest = msg.FunctionEnvironmentReloadRequest;
+                    foreach (var kv in envReloadRequest.EnvironmentVariables)
+                    {
+                        EnvironmentUtils.SetValue(kv.Key, kv.Value);
+                    }
+                    Configuration.Reload();
                     Logger.LogTrace("Specialization request received.");
```

Comment:
I would move this log to the beginning of the 'case', if ever we fail in SetValues we will have the log where it failed and specialization request was received.

Reply:
Moved it up and marked it as "Info" level log than "trace" level.